ksql-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -81,6 +87,64 @@ public void shouldBuildSourceNode() throws Exception {
     assertThat(node.topics(), equalTo("[test2]"));
   }
 
+  @Test
+  public void shouldBuildTableNodeWithCorrectAutoCommitOffsetPolicy() throws Exception {
+
+    StreamsBuilder streamsBuilder = mock(StreamsBuilder.class);
+    KsqlConfig ksqlConfig = mock(KsqlConfig.class);
+    KafkaTopicClient kafkaTopicClient = mock(KafkaTopicClient.class);
+    MetastoreUtil metastoreUtil = mock(MetastoreUtil.class);
+    FunctionRegistry functionRegistry = mock(FunctionRegistry.class);
+
+    class RightTable extends PlanNode {
+      final Schema schema;
+
+      public RightTable(final PlanNodeId id, Schema schema) {
+        super(id);
+        this.schema = schema;
+      }
+      @Override
+      public Schema getSchema() {
+        return schema;
+      }
+
+      @Override
+      public Field getKeyField() {
+        return null;
+      }
+
+      @Override
+      public List<PlanNode> getSources() {
+        return null;
+      }
+
+      @Override
+      public SchemaKStream buildStream(StreamsBuilder builder, KsqlConfig ksqlConfig,
+                                       KafkaTopicClient kafkaTopicClient,
+                                       MetastoreUtil metastoreUtil,
+                                       FunctionRegistry functionRegistry,
+                                       Map<String, Object> props) {
+        if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG) &&
+            props.get(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG).toString().equalsIgnoreCase
+                ("EARLIEST")) {
+          return mock(SchemaKTable.class);
+        } else {
+          throw new KsqlException("auto.offset.reset should be set to EARLIEST.");
```

Comment:
So if this regresses again, this exception will be thrown in the test?

Reply:
Yes, if the Stream-table join changes the table auto.offset.reset this will catch it.